### PR TITLE
[IPAD-446] Reviewer requests UI

### PR DESCRIPTION
**Pull request by paulnordlund**
1) Study report linkouts - change icon to "info"; when report exists, change icon color to orange and animate (shake)
Screen Shot 2023-06-15 at 8 42 42 PM

2) For set analysis plot, change "View Plot" to "View Set Intersection"
Screen Shot 2023-06-15 at 8 43 14 PM

3) For ALL plot dropdowns (differential dynamic plots, differential plots overlay, enrichment triple pane plot), on hover display a popup with the description (if exists - only demo study)
Screen Shot 2023-06-15 at 8 43 03 PM

4) For the differential result
s table and filtered differential results table (in enrichment triple pane) columns, on hover display an html "title" tooltip with description (if exists - only demo study)
Screen Shot 2023-06-15 at 8 42 51 PM

### CODE REVIEW

File: src/components/Differential/DifferentialSearch.jsx [hunk 1]
```diff
@@ -292,6 +296,54 @@ class DifferentialSearch extends Component {
...
   };
 
+  getResultsColumnTooltips = (study) => {
+    const { onSetDifferentialResultsColumnTooltips } = this.props;
+    cancelGetDifferentialResultsColumnTooltips();
+    let cancelToken = new CancelToken((e) => {
+      cancelGetDifferentialResultsColumnTooltips = e;
+    });
+    omicNavigatorService
+      .getResultsColumnTooltips(study)
```
Comment (paulnordlund) [inline src/components/Differential/DifferentialSearch.jsx:306]: unfortunately getting the column header tooltips requires a call (passing study)

File: src/components/Differential/DifferentialSearch.jsx [hunk 2]
```diff
@@ -292,6 +296,54 @@ class DifferentialSearch extends Component {
...
+  };
+
+  getDifferentialPlotDescriptions = (study) => {
+    const { onSetDifferentialPlotDescriptions } = this.props;
+    cancelGetDifferentialPlotDescriptions();
+    let cancelToken = new CancelToken((e) => {
+      cancelGetDifferentialPlotDescriptions = e;
+    });
+    omicNavigatorService
+      .getPlotDescriptions(study)
```
Comment (paulnordlund) [inline src/components/Differential/DifferentialSearch.jsx:331]: plot descriptions also requires a call (passing study)...

File: src/components/Differential/DifferentialSearch.jsx [hunk 3]
```diff
@@ -916,16 +968,26 @@ class DifferentialSearch extends Component {
...
           >
-            <Icon
-              name="line graph"
-              size="large"
-              className="StudyHtmlIcon"
-              inverted
-              circular
-              disabled={!differentialStudyHrefVisible}
-            />
+            <Transition
```
Comment (paulnordlund) [inline src/components/Differential/DifferentialSearch.jsx:973]: to animate the report linkout, I use this transition wrapper; there sometimes seems to be some funky behavior when hovering over the icon.  i've played with it and cannot figure out anything that always plays nicely, so if yall are seeing an issue, i'd say we drop the animation for another, or entirely

File: src/components/Differential/DifferentialSearch.jsx [hunk 4]
```diff
@@ -1020,7 +1078,7 @@ class DifferentialSearch extends Component {
           >
             <Radio
               toggle
-              label={PlotLabel}
+              label="View Plot Intersections"
```
Comment (paulnordlund) [inline src/components/Differential/DifferentialSearch.jsx:1081]: "View Plot Intersections" is quite long...I've added a window width check to change CSS (drop the toggle to a new line) if the space is too tight.  I could maybe make it a listener so it's dynamic, but that might bog it down re-rendering

File: src/components/Differential/PlotsOverlay.jsx [hunk 5]
```diff
@@ -160,7 +162,27 @@ class PlotsOverlay extends PureComponent {
...
       // );
 
-      if (this.props.differentialPlotTypes && this.props.plotOverlayData) {
+      if (differentialPlotTypes && plotOverlayData) {
+        let differentialPlotDescription = null;
+        let currentDifferentialPlotDescriptions =
+          differentialPlotDescriptions?.[differentialModel] || {};
+        const currentPlotText =
+          options?.[activeTabIndexPlotsOverlayVar]?.text || null;
+        if (!isObjectEmpty(currentDifferentialPlotDescriptions)) {
```
Comment (paulnordlund) [inline src/components/Differential/PlotsOverlay.jsx:171]: The JSON from the backend is a bit atypical, so I made this work for plots.  If we could get the data in with other plots or tests data then we wouldn't need this, but with reviewer response deadline Wednesday it has to suffice

File: src/components/Differential/DifferentialSearch.jsx (hunk 3 above)
Comment (ternst23) [reply src/components/Differential/DifferentialSearch.jsx:973]: if i hover, coming up from the bottom, the tooltip flashes for like a millisecond.

File: src/components/Differential/DifferentialSearch.jsx (hunk 4 above)
Comment (ternst23) [reply src/components/Differential/DifferentialSearch.jsx:1081]: If it drops, can it be immediately right justified as opposed to ...
image

**Review by JoeDalen81 — APPROVED** [review]

File: src/components/Differential/DifferentialSearch.jsx (hunk 3 above)
Comment (paulnordlund) [reply src/components/Differential/DifferentialSearch.jsx:973]: yes, wherever the popup opens (e.g. bottom center), that's where the hover has trouble. I cannot find a solution, so moving the popup to the top, which I think is the least likely place the user will approach the icon.  Most often I'm moving from the left or bottom

File: src/components/Differential/DifferentialSearch.jsx (hunk 4 above)
Comment (paulnordlund) [reply src/components/Differential/DifferentialSearch.jsx:1094]: @ternst23, sure, I've added a resize listener so it will now change dynamically

**Review by ternst23 — APPROVED** [review]

**Pull request merged**